integration_test/common/common.go

```diff
@@ -70,44 +69,34 @@ type ConfigurationFields struct {
 }
 
 type InputConfiguration struct {
-	Type   string                `yaml:"type" validate:"required"`
-	Fields []ConfigurationFields `yaml:"fields" validate:"required"`
+	Type   string                 `yaml:"type" validate:"required"`
+	Fields []*ConfigurationFields `yaml:"fields" validate:"required"`
 }
 
 type ConfigurationOptions struct {
-	LogsConfiguration    []InputConfiguration `yaml:"logs"`
-	MetricsConfiguration []InputConfiguration `yaml:"metrics"`
+	LogsConfiguration    []*InputConfiguration `yaml:"logs" validate:"required_without=MetricsConfiguration"`
+	MetricsConfiguration []*InputConfiguration `yaml:"metrics" validate:"required_without=LogsConfiguration"`
 }
 
 type IntegrationMetadata struct {
-	PublicUrl                    string                       `yaml:"public_url,omitempty"`
+	PublicUrl                    string                       `yaml:"public_url"`
 	ShortName                    string                       `yaml:"short_name" validate:"required"`
 	LongName                     string                       `yaml:"long_name" validate:"required"`
 	Description                  string                       `yaml:"description" validate:"required"`
-	ConfigureIntegration         string                       `yaml:"configure_integration,omitempty"`
-	ExpectedLogs                 []ExpectedLog                `yaml:"expected_logs,omitempty"`
-	ExpectedMetrics              []ExpectedMetric             `yaml:"expected_metrics,omitempty"`
-	MinimumSupportedAgentVersion MinimumSupportedAgentVersion `yaml:"minimum_supported_agent_version,omitempty"`
-	SupportedAppVersion          []string                     `yaml:"supported_app_version" validate:"required"`
-	ConfigurationOptions         ConfigurationOptions         `yaml:"configuration_options" validate:"required"`
-	RestartAfterInstall          bool                         `yaml:"restart_after_install,omitempty"`
+	ConfigurationOptions         *ConfigurationOptions        `yaml:"configuration_options" validate:"required"`
+	ConfigureIntegration         string                       `yaml:"configure_integration"`
+	ExpectedLogs                 []*ExpectedLog               `yaml:"expected_logs"`
+	ExpectedMetrics              []*ExpectedMetric            `yaml:"expected_metrics"`
+	MinimumSupportedAgentVersion MinimumSupportedAgentVersion `yaml:"minimum_supported_agent_version"`
+	SupportedAppVersion          []string                     `yaml:"supported_app_version" validate:"required,unique,min=1"`
+	RestartAfterInstall          bool                         `yaml:"restart_after_install"`
 }
 
-var validate *validator.Validate
-
 // ValidateMetrics checks that all enum fields have valid values and that
 // there is exactly one representative metric in the slice.
-func ValidateMetrics(metrics []ExpectedMetric) error {
+func ValidateMetrics(metrics []*ExpectedMetric) error {
 	var err error
 
-	// Field validation, one-by-one for better error messages
```

Comment:
What impact (e.g. before and after) does this deletion have on the resulting error messages?

Reply:
The validation got moved inside third_party_test.go.
Instead of testing them one by one, it's testing the yaml file as a integrationMetadata struct as a full set.